R/tmiic.utils.R

```diff
@@ -0,0 +1,1343 @@
+  #
+  if ( ! is.null (delta_t) )
+    {
+    if ( test_param_wrong_int (delta_t, min=1, max=NA) )
+      {
+      if ( "delta_t" %in% colnames(state_order) )
+        miic_warning ("parameters", "supplied value ", list_to_str (delta_t),
+          " for the delta t parameter is invalid,",
+          " if not NULL, it must be an integer >= 1.",
+          " This issue has no impact as the delta t is provided",
+          " in the state_order.")
+      else
+        miic_warning ("parameters", "supplied value ", list_to_str (delta_t),
+          " for the delta t parameter is invalid,",
+          " if not NULL, it must be an integer >= 1.",
+          " The delta t will be estimated from the data.")
+      }
+    else # valid delta_t
+      {
+      if ( ! ("delta_t" %in% colnames(state_order)) )
+        {
+        state_order$delta_t = delta_t
+        state_order$delta_t[state_order$is_contextual == 1] = 0
+        }
+      else # delta_t in state_order
+        {
+        na_in_so = is.na (state_order$delta_t)
+        if ( any (na_in_so) )
+          {
+          miic_warning ("parameters", "the delta t is supplied both",
+            " in the state_order and as parameter. As some values are missing",
+            " in the state_order, the parameter will be used to fill these",
+            " missing values.")
+          state_order$delta_t[na_in_so & (state_order$is_contextual == 0)] = delta_t
+          }
+        else
+          miic_warning ("parameters", "the delta t is supplied both",
+            " in the state_order and as parameter. The parameter will be",
+            " ignored.")
+        }
+      }
+    }
+  #
+  # Check movavg
+  #
+  if ( ! is.null (movavg) )
+    {
+    if (   test_param_wrong_int (movavg, min=0, max=NA)
+       || (movavg == 1) )
+      {
+      if ( "movavg" %in% colnames(state_order) )
+        miic_warning ("parameters", "supplied value ", list_to_str (movavg),
+          " for the moving average parameter is invalid,",
+          " if not NULL or 0, it must be an integer >= 2.",
+          " This issue has no impact as the moving average is provided",
+          " in the state_order.")
+      else
+        miic_warning ("parameters", "supplied value ", list_to_str (movavg),
+          " for the moving average parameter is invalid,",
+          " if not NULL or 0, it must be an integer >= 2.",
+          " The moving average parameter will be ignored.")
+      }
+    else # valid movavg
+      {
+      if ( ! ("movavg" %in% colnames(state_order)) )
+        {
+        state_order$movavg = movavg
+        # No movavg on discrete or contextual vars
+        state_order$movavg[state_order$var_type == 0] = 0
+        state_order$movavg[state_order$is_contextual == 1] = 0
+        }
+      else # movavg in state_order
+        {
+        na_in_so = is.na (state_order$movavg)
+        if ( any (na_in_so) )
+          {
+          miic_warning ("parameters", "the moving average is supplied both",
```

Comment:
EN (OD)

Reply:
is both supplied in the state_order and as a parameter.